src/panels/lovelace/badges/hui-entity-filter-badge.ts

```diff
@@ -0,0 +1,141 @@
+import { EntityFilterBadgeConfig } from "./types";
+import { evaluateFilter } from "../common/evaluate-filter";
+
+class EntityFilterBadge extends HTMLElement implements LovelaceBadge {
+  public isPanel?: boolean;
+  private _elements?: LovelaceBadge[];
+  private _config?: EntityFilterBadgeConfig;
+  private _configEntities?: EntityFilterEntityConfig[];
+  private _hass?: HomeAssistant;
+  private _oldEntities?: EntityFilterEntityConfig[];
+
+  public setConfig(config: EntityFilterBadgeConfig): void {
+    if (!config.entities || !Array.isArray(config.entities)) {
+      throw new Error("entities must be specified.");
+    }
+
+    if (
+      !(config.state_filter && Array.isArray(config.state_filter)) &&
+      !config.entities.every(
+        (entity) =>
+          typeof entity === "object" &&
+          entity.state_filter &&
+          Array.isArray(entity.state_filter)
+      )
+    ) {
+      throw new Error("Incorrect filter config.");
+    }
+
+    this._config = config;
+    this._configEntities = undefined;
+
+    if (this.lastChild) {
+      this.removeChild(this.lastChild);
+      this._elements = undefined;
+    }
+  }
+
+  set hass(hass: HomeAssistant) {
+    if (!hass || !this._config) {
+      return;
+    }
+
+    if (!this.haveEntitiesChanged(hass)) {
+      this._hass = hass;
+      return;
+    }
+
+    this._hass = hass;
+
+    if (!this._configEntities) {
+      this._configEntities = processConfigEntities(this._config.entities);
+    }
+
+    const entitiesList = this._configEntities.filter((entityConf) => {
+      const stateObj = hass.states[entityConf.entity];
+
+      if (!stateObj) {
+        return false;
+      }
+
+      if (entityConf.state_filter) {
+        for (const filter of entityConf.state_filter) {
+          if (evaluateFilter(stateObj, filter)) {
+            return true;
+          }
+        }
+      } else {
+        for (const filter of this._config!.state_filter) {
+          if (evaluateFilter(stateObj, filter)) {
+            return true;
+          }
+        }
+      }
+
+      return false;
+    });
+
+    if (entitiesList.length === 0) {
+      this.style.display = "none";
+      return;
+    }
+
+    const isSame =
+      this._oldEntities &&
+      entitiesList.length === this._oldEntities.length &&
+      entitiesList.every((entity, idx) => entity === this._oldEntities![idx]);
+
+    if (!isSame) {
+      this._elements = [];
+      for (const badgeConfig of entitiesList) {
+        const element = createBadgeElement(badgeConfig);
+        this._elements.push(element);
+      }
+      this._oldEntities = entitiesList;
+    }
+
+    if (!this._elements) {
+      return;
+    }
+
+    for (const element of this._elements) {
+      element.hass = hass;
+    }
+
+    // Attach element if it has never been attached.
+    if (!this.lastChild) {
+      for (const element of this._elements) {
+        this.appendChild(element);
+      }
+    }
+
+    this.style.display = "inline";
```

Comment:
Isn't this the default?

Reply:
yes, but I set it to `none` if no entities match so this will reset it